packages/matter-node-ble.js/src/ble/NobleBleChannel.ts

```diff
@@ -0,0 +1,238 @@
+
+const logger = Logger.get("BleChannel");
+
+/**
+ * Convert a UUID in noble's format to a proper UUID.
+ *
+ * @param {string} uuid - UUID to convert
+ * @returns {string} UUID
+ */
+function nobleUuidToUuid(uuid: string): string {
+    uuid = uuid.toUpperCase();
+
+    if (uuid.length !== 32) {
+        return uuid;
+    }
+
+    const parts = [
+        uuid.substring(0, 8),
+        uuid.substring(8, 12),
+        uuid.substring(12, 16),
+        uuid.substring(16, 20),
+        uuid.substring(20, 32),
+    ];
+
+    return parts.join('-');
+}
+
+export class NobleBleCentralInterface implements NetInterface {
+    private openChannels: Map<ServerAddress, Peripheral> = new Map();
+    private onMatterMessageListener: ((socket: Channel<ByteArray>, data: ByteArray) => void) | undefined;
+
+    async openChannel(address: ServerAddress): Promise<Channel<ByteArray>> {
+        if (address.type !== "ble") {
+            throw new Error(`Unsupported address type ${address.type}.`);
+        }
+        if (this.onMatterMessageListener === undefined) {
+            throw new Error(`Network Interface was not added to the system yet.`);
+        }
+
+        // Get the peripheral by address and connect to it.
+        const { peripheral, hasAdditionalAdvertisementData } = (Ble.get().getBleScanner() as BleScanner).getDiscoveredDevice(address.peripheralAddress);
+        if (peripheral.state === "connected" || peripheral.state === "connecting") {
+            throw new Error(`Peripheral ${address.peripheralAddress} is already connected or connecting. Only one connection supported right now.`);
+        }
+        if (this.openChannels.has(address)) {
+            throw new Error(`Peripheral ${address.peripheralAddress} is already connected. Only one connection supported right now.`);
+        }
+        await peripheral.connectAsync();
+
+        // Once the peripheral has been connected, then discover the services and characteristics of interest.
+        const services = await peripheral.discoverServicesAsync([BLE_MATTER_SERVICE_UUID]);
+
+        for (const service of services) {
+            logger.debug(`found service: ${service.uuid}`);
+            if (service.uuid !== BLE_MATTER_SERVICE_UUID) continue;
+
+            // So, discover its characteristics.
+            const characteristics = await service.discoverCharacteristicsAsync();
+
+            let characteristicC1ForWrite: Characteristic | undefined;
+            let characteristicC2ForSubscribe: Characteristic | undefined;
+            let additionalCommissioningRelatedData: ByteArray | undefined;
+
+            for (const characteristic of characteristics) {
+                // Loop through each characteristic and match them to the UUIDs that we know about.
+                logger.debug('found characteristic:', characteristic.uuid, characteristic.properties);
+
+                switch (nobleUuidToUuid(characteristic.uuid)) {
+                    case BLE_MATTER_C1_CHARACTERISTIC_UUID:
```

Comment:
Would be helpful for comprehension to use e.g. ClientTxBuffer, ClientRxBuffer, AdditionalCommissioningData instead of C1, C2, C3

Reply:
I decided to callit C1, C2 and such because also specs use this "In test descriptions" and in fact this makes it easiert to map stuff to the specs while understanding/debugging. Else you always need one "name-mapping hop"